src/process/wait.rs

```diff
@@ -99,9 +112,145 @@ impl WaitStatus {
+impl WaitidStatus {
+    /// Returns whether the process is currently stopped.
+    #[inline]
+    pub fn stopped(&self) -> bool {
+        self.si_code() == backend::c::CLD_STOPPED
+    }
+
+    /// Returns whether the process is currently trapped.
+    #[inline]
+    pub fn trapped(&self) -> bool {
+        self.si_code() == backend::c::CLD_TRAPPED
+    }
+
+    /// Returns whether the process has exited normally.
+    #[inline]
+    pub fn exited(&self) -> bool {
+        self.si_code() == backend::c::CLD_EXITED
+    }
+
+    /// Returns whether the process was terminated by a signal
+    /// and did not create a core file.
+    #[inline]
+    pub fn killed(&self) -> bool {
+        self.si_code() == backend::c::CLD_KILLED
+    }
+
+    /// Returns whether the process was terminated by a signal
+    /// and did create a core file.
+    #[inline]
+    pub fn dumped(&self) -> bool {
+        self.si_code() == backend::c::CLD_DUMPED
+    }
+
+    /// Returns whether the process has continued from a job control stop.
+    #[inline]
+    pub fn continued(&self) -> bool {
+        self.si_code() == backend::c::CLD_CONTINUED
+    }
+
+    /// Returns the number of the signal that stopped the process,
+    /// if the process was stopped by a signal.
+    #[inline]
+    #[cfg(not(any(target_os = "netbsd", target_os = "fuchsia", target_os = "emscripten")))]
+    pub fn stopping_signal(&self) -> Option<u32> {
+        if self.stopped() {
+            Some(self.si_status() as _)
+        } else {
+            None
+        }
+    }
+
+    /// Returns the number of the signal that trapped the process,
+    /// if the process was trapped by a signal.
+    #[inline]
+    #[cfg(not(any(target_os = "netbsd", target_os = "fuchsia", target_os = "emscripten")))]
+    pub fn trapping_signal(&self) -> Option<u32> {
+        if self.trapped() {
+            Some(self.si_status() as _)
+        } else {
+            None
+        }
+    }
+
+    /// Returns the exit status number returned by the process,
+    /// if it exited normally.
+    #[inline]
+    #[cfg(not(any(target_os = "netbsd", target_os = "fuchsia", target_os = "emscripten")))]
+    pub fn exit_status(&self) -> Option<u32> {
+        if self.exited() {
+            Some(self.si_status() as _)
+        } else {
+            None
+        }
+    }
+
+    /// Returns the number of the signal that terminated the process,
+    /// if the process was terminated by a signal.
+    #[inline]
+    #[cfg(not(any(target_os = "netbsd", target_os = "fuchsia", target_os = "emscripten")))]
+    pub fn terminating_signal(&self) -> Option<u32> {
+        if self.killed() || self.dumped() {
+            Some(self.si_status() as _)
+        } else {
+            None
+        }
+    }
+
+    /// Returns a reference to the raw platform-specific `siginfo_t` struct.
+    #[inline]
+    pub const fn as_raw(&self) -> &backend::c::siginfo_t {
+        &self.0
+    }
+
+    #[cfg(linux_raw)]
```

Comment:
Rustix does have a few places where the public API for `linux_raw` differs from the public API for the libc backend on Linux, but we're working on eliminating them, as it makes code less portable. Is it complicated to add Linux support using the libc backend as well? Also, yeah, your suggestion to push these details into src/backend/* sounds good; perhaps in new src/backend/*/wait.rs files?

Reply:
It should already work with the libc backend (CI should test this, right?) — it's the final `not(any(…))` version. The libc crate does make an effort to offer the same interface here on all platforms, just lacking in a few places such as https://github.com/rust-lang/libc/pull/3185 and OpenBSD/NetBSD currently.

Reply:
Ah, I see now. I was confused by the `cfg`s :-).